core/kernel/ldelf_syscalls.c

```diff
@@ -181,13 +186,18 @@ TEE_Result ldelf_syscall_open_bin(const TEE_UUID *uuid, size_t uuid_size,
 	h = handle_get(&sys_ctx->db, binh);
 	if (h < 0)
 		goto err_oom;
-	*handle = h;
+
+	PUT_USER(h, handle);
```

Comment:
Remove `vm_check_access_rights()` at line 120 as `PUT_USER()` already takes care of this sanity test.